### PR TITLE
Update AWS SQS autoscaler to use TriggerAuthentication

### PR DESCRIPTION
**Pull request by zach-dunton-sf**
Relates to https://github.com/kedacore/keda/issues/332

### CODE REVIEW

**Comment by zach-dunton-sf** [comment]
Docs PR is here https://github.com/kedacore/keda-docs/pull/15

**Comment by tomkerkhove** [comment]
Thanks for updating the examples & specs but you should actually no longer update them as they are being removed as part of https://github.com/kedacore/keda/pull/451.

Everything is already in the docs and your other PR fixes that, thanks!

File: pkg/scalers/aws_sqs_queue_scaler.go [hunk 1]
```diff
@@ -89,39 +89,21 @@ func parseAwsSqsQueueMetadata(metadata, resolvedEnv map[string]string) (*awsSqsQ
 		return nil, fmt.Errorf("no queueURL given")
 	}
 
-	var keyName string
-	if keyName = metadata["awsAccessKeyID"]; keyName == "" {
```
Comment (tomkerkhove) [inline pkg/scalers/aws_sqs_queue_scaler.go]: Let's hold off with removing this based on #453

Comment (zach-dunton-sf) [reply pkg/scalers/aws_sqs_queue_scaler.go]: I fixed this and refactored the aws stuff a bit. I can apply it to the cloudwatch scaler also once this is merged.

Comment (tomkerkhove) [reply pkg/scalers/aws_sqs_queue_scaler.go]: I think we're good now and this has been moved to `pkg/scalers/aws_iam_authorization.go`, right?

Comment (zach-dunton-sf) [reply pkg/scalers/aws_sqs_queue_scaler.go]: yes

File: pkg/scalers/aws_sqs_queue_scaler.go [hunk 2]
```diff
@@ -21,37 +22,33 @@ import (
...
 		return nil, fmt.Errorf("Error parsing SQS queue metadata: %s", err)
 	}
 
 	sess := session.Must(session.NewSession())
 	if sess != nil {
 		s, err := session.NewSession(&aws.Config{
-			Credentials: credentials.NewStaticCredentials(meta.awsAccessKeyID, meta.awsSecretAccessKey, meta.awsSessionToken),
+			Credentials: credentials.NewStaticCredentials(meta.awsAuthorization.awsAccessKeyID, meta.awsAuthorization.awsSecretAccessKey, meta.awsAuthorization.awsSessionToken),
 			Region:      aws.String(meta.awsRegion),
 		})
```
Comment (ahmelsayed) [inline pkg/scalers/aws_sqs_queue_scaler.go]: It looks like this whole section can be removed, right?

Comment (zach-dunton-sf) [reply pkg/scalers/aws_sqs_queue_scaler.go]: It needs to stay. We have the credentials, now we need to use them. In the case of the role, we need to get temporary credentials and then use them.

Comment (zach-dunton-sf) [reply pkg/scalers/aws_sqs_queue_scaler.go]: After re-reading. I see your point.

**Pull request merged**